### ISSUE TITLE
Create build Dockerfile

### ISSUE THREAD

**Issue by echus** [ACCESS-NRI/oasis3-mct#6]

**Comment by aidanheerdegen** [comment]
I guess the easiest path is to use a spack docker container as a base

https://spack.readthedocs.io/en/latest/containers.html

I don't have a preference for which to use, but note that CentOS Stream is continuously updated, which is not a great option for a CI project. Something new but stable is better, I'd guess Ubuntu 22.04 probably fits that bill.

Doesn't look like it has any packages pre-installed though:

https://hub.docker.com/layers/spack/ubuntu-jammy/latest/images/sha256-082d0e923589a988b094e14d76707c4f672ec8b0139b1fd33100c3d412844b4c?context=explore

So first up [install a compiler](https://spack.readthedocs.io/en/latest/getting_started.html#build-your-own-compiler). Here are the supported versions:
```
$ spack info gcc
AutotoolsPackage:   gcc
         
Description:                                                                                                                                                
    The GNU Compiler Collection includes front ends for C, C++, Objective-C,                                                                                     Fortran, Ada, and Go, as well as libraries for these languages.

Homepage: https://gcc.gnu.org                                                                                                                               
                                                                                                                                                            
Preferred version:                                                                                                                                          
    12.2.0    https://ftpmirror.gnu.org/gcc/gcc-12.2.0/gcc-12.2.0.tar.xz                                                                                                  
Safe versions:                                                        
    master    [git] git://gcc.gnu.org/git/gcc.git on branch master
    12.2.0    https://ftpmirror.gnu.org/gcc/gcc-12.2.0/gcc-12.2.0.tar.xz                                                                                    
    12.1.0    https://ftpmirror.gnu.org/gcc/gcc-12.1.0/gcc-12.1.0.tar.xz                                                                                    
    11.3.0    https://ftpmirror.gnu.org/gcc/gcc-11.3.0/gcc-11.3.0.tar.xz
    11.2.0    https://ftpmirror.gnu.org/gcc/gcc-11.2.0/gcc-11.2.0.tar.xz
    11.1.0    https://ftpmirror.gnu.org/gcc/gcc-11.1.0/gcc-11.1.0.tar.xz
    10.4.0    https://ftpmirror.gnu.org/gcc/gcc-10.4.0/gcc-10.4.0.tar.xz                                                                                    
    10.3.0    https://ftpmirror.gnu.org/gcc/gcc-10.3.0/gcc-10.3.0.tar.xz                                                                                    
    10.2.0    https://ftpmirror.gnu.org/gcc/gcc-10.2.0/gcc-10.2.0.tar.xz                                                                                    
    10.1.0    https://ftpmirror.gnu.org/gcc/gcc-10.1.0/gcc-10.1.0.tar.xz                                                                                    
    9.5.0     https://ftpmirror.gnu.org/gcc/gcc-9.5.0/gcc-9.5.0.tar.xz
    9.4.0     https://ftpmirror.gnu.org/gcc/gcc-9.4.0/gcc-9.4.0.tar.xz
    9.3.0     https://ftpmirror.gnu.org/gcc/gcc-9.3.0/gcc-9.3.0.tar.xz                                                                                      
    9.2.0     https://ftpmirror.gnu.org/gcc/gcc-9.2.0/gcc-9.2.0.tar.xz
    9.1.0     https://ftpmirror.gnu.org/gcc/gcc-9.1.0/gcc-9.1.0.tar.xz
    8.5.0     https://ftpmirror.gnu.org/gcc/gcc-8.5.0/gcc-8.5.0.tar.xz
    8.4.0     https://ftpmirror.gnu.org/gcc/gcc-8.4.0/gcc-8.4.0.tar.xz
    8.3.0     https://ftpmirror.gnu.org/gcc/gcc-8.3.0/gcc-8.3.0.tar.xz
    8.2.0     https://ftpmirror.gnu.org/gcc/gcc-8.2.0/gcc-8.2.0.tar.xz
    8.1.0     https://ftpmirror.gnu.org/gcc/gcc-8.1.0/gcc-8.1.0.tar.xz
    7.5.0     https://ftpmirror.gnu.org/gcc/gcc-7.5.0/gcc-7.5.0.tar.xz
    7.4.0     https://ftpmirror.gnu.org/gcc/gcc-7.4.0/gcc-7.4.0.tar.xz
    7.3.0     https://ftpmirror.gnu.org/gcc/gcc-7.3.0/gcc-7.3.0.tar.xz
    7.2.0     https://ftpmirror.gnu.org/gcc/gcc-7.2.0/gcc-7.2.0.tar.xz
    7.1.0     https://ftpmirror.gnu.org/gcc/gcc-7.1.0/gcc-7.1.0.tar.bz2
```

`gfortran` is installed automatically with `gcc`. Start with the most recent.

Given the dependencies in https://github.com/ACCESS-NRI/oasis3-mct/issues/7#issuecomment-1324451253 the magic `spack` sauce to install should be something like

```
spack install openmpi@4.0.2 netcdf-fortran@4.5.2
```

Note that this is just specifying the `netcdf-fortran` package which will install `netcdf` as a dependency. The versions of the c and fortran netcdf packages are not the same, I found which was installed on `gadi` like so:
```
$ module load netcdf/4.7.4
$ nf-config --all

This netCDF-Fortran 4.5.2 has been built with the following features: 

  --cc        -> gcc
  --cflags    -> -I/apps/netcdf/4.7.4/include/ 

  --fc        -> gfortran
  --fflags    -> -I/apps/netcdf/4.7.4/include/GNU -I
  --flibs     -> -L/apps/netcdf/4.7.4/lib/GNU -lnetcdff  -lnetcdf -lnetcdf -ldl -lm 
  --has-f90   -> 
  --has-f03   -> yes

  --has-nc2   -> yes
  --has-nc4   -> yes

  --prefix    -> /apps/netcdf/4.7.4
  --includedir-> /apps/netcdf/4.7.4/include
  --version   -> netCDF-Fortran 4.5.2

```

**Comment by aidanheerdegen** [comment]
Some more notes (sorry for the information avalanche)

They seem to indicate it is a good idea to install `apt-get install gfortran`

https://spack.readthedocs.io/en/latest/containers.html#mpi

I guess that is a boot-strapping issue for OpenMPI?

**Comment by echus** [comment]
Closed by #8
